File: crates/vexus-index/tests/fixtures/java/Sample.java
package shop;

import java.util.List;

public class Order {
    private final List<String> lines;

    public Order(List<String> lines) {
        this.lines = lines;
    }

    // Total line count.
    public int lineCount() {
        return lines.size();
    }

    public String summary(boolean verbose) {
        var count = lineCount();
        return verbose ? new Description(count).render() : String.valueOf(count);
    }
}

interface Renderable {
    String render();
}

enum Status {
    OPEN,
    SHIPPED
}

record Description(int count) implements Renderable {
    public String render() {
        return "order with " + count + " lines";
    }
}

File: crates/vexus-index/tests/fixtures/php/sample.php
<?php
namespace Shop\Billing;

use Shop\Util\Slug;

const TAX_RATE = 8;

// Formats a cent amount for display.
function format_cents(int $cents): string {
    return "$" . strval($cents / 100);
}

class Invoice {
    private array $lines;

    public function __construct(array $lines) {
        $this->lines = $lines;
    }

    // Total with tax.
    public function total(int $tax): int {
        return $this->subtotal() + $tax;
    }

    private function subtotal(): int {
        return array_sum($this->lines);
    }
}

interface Renderable {
    public function render(): string;
}

trait Loggable {}

enum Status {
    case Open;
    case Paid;
}
